File: lester/examples/highlyrated_books/learn_with_sparkml.py
from lester.context import train_model, EstimatorTransformerDialect
from pyspark.ml.classification import LogisticRegression


@train_model(dialect=EstimatorTransformerDialect.SPARKML)
def logreg_with_fixed_hyperparams():
    return LogisticRegression(maxIter=10, regParam=0.001)

File: src/main/scala/berlin/deem/lester/example/highlyratedbooks/LearnWithSparkML.scala
package berlin.deem.lester.example.highlyratedbooks

import berlin.deem.lester.context.TrainModel
import org.apache.spark.ml.{Pipeline, PipelineStage}
import org.apache.spark.ml.classification.LogisticRegression

import scala.collection.mutable.ArrayBuffer

class LearnWithSparkML {

  @TrainModel
  def logreg(): Pipeline = {
    val nextStages = new ArrayBuffer[PipelineStage]

    nextStages += new LogisticRegression()
      .setMaxIter(10)
      .setRegParam(0.001)

    val model = new Pipeline()
      .setStages(nextStages.toArray)

    model
  }
}
